pkg/healthcheck/healthcheck.go

```diff
@@ -0,0 +1,54 @@
+package healthcheck
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/jim-minter/azure-helm/pkg/api"
+	"github.com/jim-minter/azure-helm/pkg/config"
+)
+
+func HealthCheck(ctx context.Context, m *api.Manifest, c *config.Config) error {
+	return waitForConsole(ctx, m, c)
+}
+
+func waitForConsole(ctx context.Context, m *api.Manifest, c *config.Config) error {
+	pool := x509.NewCertPool()
+	pool.AddCert(c.CaCert)
+
+	cli := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs: pool,
+			},
+		},
+		Timeout: 10 * time.Second,
+	}
+
+	req, err := http.NewRequest("HEAD", "https://"+m.PublicHostname+"/console/", nil)
+	if err != nil {
+		return err
+	}
+	req = req.WithContext(ctx)
```

Comment:
context must be non-nil here, do we want to provide a default just in case?

Reply:
https://golang.org/pkg/context/: "Do not pass a nil Context, even if a function permits it. Pass context.TODO if you are unsure about which Context to use."

Reply:
so if/when an issue arises with this we'll just point them to go doc?  Seems dodgy

Reply:
Agreed on defaulting to TODO if we get a nil context.

Reply:
I think you're misinterpreting the documentation.  

> "Do not pass a nil Context, even if a function permits it. Pass context.TODO if you are unsure about which Context to use."

These are directions to the caller - MSFT's RP.  The "even if a function permits it" indicate it is perfectly valid to reject the call to `HealthCheck` if a nil context is given or default a timeout ourselves.  

You're defaulting to a runtime panic :smile: 

```
func (r *Request) WithContext(ctx context.Context) *Request {
	if ctx == nil {
		panic("nil context")
	}
```